pkgs/development/haskell-modules/make-package-set.nix

```
@@ -552,7 +552,13 @@ in package-set { inherit pkgs lib callPackage; } self // {
       pkgs.runCommandLocal name
         {
           inherit src;
-          nativeBuildInputs = [ buildHaskellPackages.cabal-install ];
+          nativeBuildInputs = [
+            buildHaskellPackages.cabal-install
+
+            # TODO after https://github.com/haskell/cabal/issues/8352
+            #      remove ghc
+            buildHaskellPackages.ghc
```

The `ghc` attribute is a bit like `stdenv`, the package set re-exposes the GHC it uses to build with. The bootstrap set compiler may be of a different version.

```suggestion
            self.ghc
```